onnxruntime/test/providers/provider_test_utils.h

```diff
@@ -149,6 +149,23 @@ struct TTensorType {
 template <typename T>
 const TTypeProto<T> TTensorType<T>::s_type_proto;
 
+struct TSparseTensorProto {
```

Comment:
Why move away from the pattern set by TTypeProto ?

Reply:
This is because it allows less templating and have non-templated implementaiton. I am planning to re-work the rest.